swiftinterface compilation does not re-serialise static library builds

### Description

It seems that I recently started getting link errors related to `swiftCxx.lib` and `swiftCxxStdlib.lib` such as:

Windows (x64/arm64):
```
lld-link: error: could not open 'swiftCxx.lib': no such file or directory
lld-link: error: could not open 'swiftCxxStdlib.lib': no such file or directory
```
```
LINK : fatal error LNK1104: cannot open file 'swiftCxx.lib'
```
where I presume it should probably need to load instead `libswiftCxx.lib` and `libswiftCxxStdlib.lib`  which actually exist in the toolchain directory, as opposed to `swiftCxx.lib` and `swiftCxxStdlib.lib` which don't exist there.

Locally reverting https://github.com/swiftlang/swift/pull/77559 (and https://github.com/swiftlang/swift-installer-scripts/pull/344) seems to make this error go away.

### Reproduction

On a recent toolchain build in the main branch:

Windows
```
C:\Users\hiroshi\Test>cat Test.swift
import CXXModule

C:\Users\hiroshi\Test>cat Inputs\header.h
#ifndef HEADER_H
#define HEADER_H

#endif // HEADER_H

C:\Users\hiroshi\Test>cat Inputs\module.modulemap
module CXXModule {
  header "header.h"
  requires cplusplus
}

C:\Users\hiroshi\Test>swiftc Test.swift -cxx-interoperability-mode=default -o Test.exe -I C:\Users\hiroshi\SwiftCxxTest\Inputs
error: link command failed with exit code 1104 (use -v to see invocation)
LINK : fatal error LNK1104: cannot open file 'swiftCxx.lib'
clang: error: linker command failed with exit code 1104 (use -v to see invocation)
```

### Expected behavior

Successful linking, no link errors.

### Environment

With a recent main-branch toolchain build on windows.

### Additional information

_No response_

--- a/lib/Frontend/ModuleInterfaceBuilder.cpp
+++ b/lib/Frontend/ModuleInterfaceBuilder.cpp
@@ -279,6 +279,7 @@ std::error_code ExplicitModuleInterfaceBuilder::buildSwiftModuleFromInterface(
   // optimization pipeline.
   SerializationOptions SerializationOpts;
   std::string OutPathStr = OutputPath.str();
+  SerializationOpts.StaticLibrary = FEOpts.Static;
   SerializationOpts.OutputPath = OutPathStr.c_str();
   SerializationOpts.ModuleLinkName = FEOpts.ModuleLinkName;
   SerializationOpts.AutolinkForceLoad =

--- a/test/ModuleInterface/Inputs/SR77756/dynamic.swiftinterface
+++ b/test/ModuleInterface/Inputs/SR77756/dynamic.swiftinterface
@@ -1,0 +1,7 @@
+// swift-interface-format-version: 1.0
+// swift-module-flags: -module-name dynamic
+
+import Swift
+
+public func f() {}
+

--- a/test/ModuleInterface/Inputs/SR77756/static.swiftinterface
+++ b/test/ModuleInterface/Inputs/SR77756/static.swiftinterface
@@ -1,0 +1,6 @@
+// swift-interface-format-version: 1.0
+// swift-module-flags: -static -module-name static
+
+import Swift
+
+public func f() {}

--- a/test/ModuleInterface/swift-interface-compile.swift
+++ b/test/ModuleInterface/swift-interface-compile.swift
@@ -1,0 +1,14 @@
+// RUN: %empty-directory(%t)
+// RUN: %target-swift-frontend -module-cache-path %t/mcp -I %S/Inputs/SR77756 -c %s -o /dev/null -D STATIC
+// RUN: llvm-bcanalyzer -dump %t/mcp/static-*.swiftmodule | %FileCheck %s -check-prefix CHECK-STATIC
+// RUN: %target-swift-frontend -module-cache-path %t/mcp -I %S/Inputs/SR77756 -c %s -o /dev/null
+// RUN: llvm-bcanalyzer -dump %t/mcp/dynamic-*.swiftmodule | %FileCheck %s -check-prefix CHECK-DYNAMIC
+
+#if STATIC
+import `static`
+#else
+import `dynamic`
+#endif
+
+// CHECK-STATIC: IS_STATIC
+// CHECK-DYNAMIC-NOT: IS_STATIC

